libp2p/NodeTable.cpp

```diff
@@ -593,6 +618,29 @@ void NodeTable::doDiscovery()
     });
 }
 
+void NodeTable::doHandleTimeouts()
+{
+    m_timers.schedule(c_handleTimeoutsIntervalMs, [this](boost::system::error_code const& _ec) {
+        if ((_ec && _ec.value() == boost::asio::error::operation_aborted) || m_timers.isStopped())
+            return;
+
+        for (auto it = m_sentPings.begin(); it != m_sentPings.end();)
+        {
+            if (chrono::steady_clock::now() > it->second.first + DiscoveryDatagram::c_timeToLive)
+            {
+                if (auto node = nodeEntry(it->first))
+                    dropNode(node);
```

Comment:
`dropNode()` looks to be quite inefficient by requiring `shared_ptr`. Still you can do `std::move()`.